device/core/src/client_diagnostic.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { Message, errors } from 'azure-iot-common';
+
+/**
+ * Data structure to store diagnostic property data
+ */
+export class DiagnosticPropertyData {
+  // Key of creation time in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_CREATION_TIME_UTC_PROPERTY: string = 'creationtimeutc';
+  // Equal symbol in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_SYMBOL_EQUAL: string = '=';
+  // And symbol in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_SYMBOL_AND: string = '&';
+
+  private diagnosticId: string;
+  private diagnosticCreationTimeUtc: string;
+
+  /**
+   * @constructor
+   */
+  constructor(diagnosticId: string, diagnosticCreationTimeUtc: string) {
```

Comment:
>diagnosticId [](start = 14, length = 12)

"diagnostic" prefix can be guessed from the class name and is unnecessary too

Reply:
fixed